pkg/driver/bigquery/bigquery.go

```diff
@@ -527,18 +682,55 @@ func tableExists(ctx context.Context, client *bigquery.Client, datasetID, tableN
 	return false, err
 }
 
-func connectionString(url *url.URL) string {
-	c, _ := configFromURI(url)
+func connectionString(u *url.URL) string {
+	var params, locationPath string
+
+	fields := strings.Split(strings.TrimPrefix(u.Path, "/"), "/")
+	projectID := u.Hostname()
+	dataset := fields[len(fields)-1]
```

Comment:
The `projectID` should be `fields[0]`, not the `u.Hostname` for dbmate, and `dataset` should be `fields[1]`, therefore `len(fields) == 2` must be true, otherwise we should return an error -- but, unfortunately we would have to adjust a bit of tode to change the return signature from just `(string)` to `(string, error)` -- but, maybe it would be worth doing since `connectionString()` is called from only one place in the driver code, and only one place in the tests?

Reply:
i have changed the logic a bit to use `net/url` but the tests remain the same so it should work as expected

Reply:
Once we get through the current batch of changes, I'll provide additional tests for `connectionString()`, as there's a lot of cases that aren't being tested.